src/ray/util/logging.cc

```diff
@@ -76,18 +76,24 @@ std::ostream &operator<<(std::ostream &os, const StackTrace &stack_trace) {
   void *frames[MAX_NUM_FRAMES];
 
 #ifndef _WIN32
+  // A deleter can be used with std::unique_ptr to free memory without passing function
+  // pointer of free
+  struct FreeDeleter {
+    void operator()(void *ptr) const { free(ptr); }
```

Comment:
This is bad, `unique_ptr` uses `new` internally, which should be paired with `delete` rather than `free`.